### PR TITLE
Delete variations from the single page does not reflect correctly in the parent product

### LINKED ISSUE #40897: Delete variations from the single page does not reflect correctly in the parent product
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

If the parent product has a list of variations with more than one page in the variations table and when ever a variation is removed from using the single variation page then the variations table from the parent product receives the update but the current table page shows one less variation instead of making a request to fully get the real page with the right amount of items.

https://github.com/woocommerce/woocommerce/assets/13334210/74666fa9-fa47-4484-ae0f-975e8966c526

**ACs**

- [ ] Fix the amount of variations shown per page in the variations table of the parent product after removing a variation from the single variation page

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #40897

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-variation-management` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and add enough attributes to generate 6 variations
4. Wait until variations get generated
5. Then publish the product
6. Click `Edit` button from the `Variations` table to edit any variation listed there.
7. You should be redirected to the single variation page to edit it.
8. On the top right corner of the screen the more menu should be shown with a new item called `Delete variation` 
image

9. When click it, e confirmation modal should be shown 
image

10. I you click delete button from the modal then right after the variation is deleted a notice should be shown at the bottom left corner of the screen saying the that the variation was deleted and you should be redirected to the variation parent product page.
11. Then go to the parent product using the top left back button on the screen.
12. From the parent product page and under the `Variations` table, the amount of items should be exactly 5.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/hooks/use-variation-switcher.ts [hunk 1]
```diff
@@ -66,6 +72,7 @@ export function useVariationSwitcher( {
 			parentProductType || 'product',
 			parentId,
 		] );
+		invalidateResolutionForStoreSelector( 'getProductVariations' );
```
Comment (octaedro) [inline packages/js/product-editor/src/hooks/use-variation-switcher.ts:77]: I think that we should also invalidate the `getProductVariationsTotalCount` resolution as well, to update the variation without price count.

Screenshot 2023-10-20 at 11 32 36

**Review by octaedro — COMMENTED** [review]
Nice job @mdperez86! The code looks mainly good, I just left a small comment.

File: packages/js/product-editor/src/hooks/use-variation-switcher.ts (hunk 1 above)
Comment (mdperez86) [reply packages/js/product-editor/src/hooks/use-variation-switcher.ts:77]: done -> https://github.com/woocommerce/woocommerce/pull/40900/commits/1838e63fafab328620ab78b46518197953b262f3

**Review by octaedro — APPROVED** [review]
LGTM! 🚀

**Pull request merged**